src/java/org/apache/cassandra/dht/IPartitioner.java

```diff
@@ -128,8 +128,9 @@ default Token getMaximumToken()
     /**
      * Abstract type that orders the same way as DecoratedKeys provided by this partitioner.
      * Used by secondary indices.
+     * @param baseType base type for PartitionerDefinedOrder
      */
-    public AbstractType<?> partitionOrdering();
+    public AbstractType<?> partitionOrdering(AbstractType<?> baseType);
```

Comment:
I'm not sure whether the partitioner can be considered a public API, given that one can supply a custom class. If this is so I understand that we should get consensus on the mail list to change its API, even in a major (and I think we are officially in 4.2).

A workaround to keep the API compatibility, if needed, would be providing a default implementation of the new `partitionOrdering(AbstractType)`, and keeping the previous `partitionOrdering()` method.

@blambov wdyt?

Reply:
>  A workaround to keep the API compatibility, if needed, would be providing a default implementation of the new `partitionOrdering(AbstractType)`, and keeping the previous `partitionOrdering()` method.

This works for me. Would it need mailing list agreement too?

Reply:
I think it wouldn't, since we wouldn't be breaking any public API compatibility. For context on the discussions about releasing 5.0 and API changes, see:

- https://www.mail-archive.com/dev@cassandra.apache.org/msg19516.html
- https://www.mail-archive.com/dev@cassandra.apache.org/msg19925.html
- https://issues.apache.org/jira/browse/CASSANDRA-17973

Reply:
That's what I mean with the workaround:
```java
@Deprecated // use #partitionOrdering(AbstractType) instead, see CASSANDRA-17698 for details
AbstractType<?> partitionOrdering();

default AbstractType<?> partitionOrdering(@Nullable AbstractType<?> partitionKeyType)
{
    return partitionOrdering();
}
```
And then the proper partitioner implementations can override `partitionOrdering(AbstractType)`.

Reply:
ok

Reply:
AbstractType<?> partitionOrdering();
I will just mark this method Deprecated and left the comment, but I will not remove the public flag of the method , though the public is useless.